### ISSUE TITLE
Variant of no-unused-expressions that allows type assertions?

### ISSUE THREAD

**Issue by mkscrg** [gajus/eslint-plugin-flowtype#271]
It's sometimes helpful to write a bare type assertion with no side-effects. For example when discriminating a disjoint union in a switch:

```js
type Action
  = { type: 'FOO', doFoo(_: number): void }
  | { type: 'BAR', doBar(_: string): void };

function handleAction(act: Action): void {
  switch (act.type) {
    case 'FOO':
      doFoo(10);
      break;
    case 'BAR':
      doBar('hello');
      break;
    default:
      (act: empty); // type error unless we've exhausted the possibilities for `act.type`
  }
}
```

Unfortunately these are caught by ESLint's `no-unused-expressions` rule.

Is there a clean way to "fix" this with `eslint-plugin-flowtype`? Maybe we could expose a `flowtype/no-unused-expressions` that additionally ignores type assertions?

**Comment by mkscrg** [comment]
Scratched this itch with #272

**Comment by bradennapier** [comment]
Love this one! Thanks!

**Comment by cinic** [comment]
Not working for me.

Eslint: 4.14
Eslint-plugin-flowtype: 2.44.0
babel-eslint: 8.2.1

Eslint error: Expected an assignment or function call and instead saw an expression. (no-unused-expressions).

**Comment by pronebird** [comment]
Hey, this doesn't work for me either, my eslintrc:

```
{
  "extends": [
    "eslint:recommended",
    "plugin:react/recommended",
    "plugin:promise/recommended",
    "plugin:flowtype/recommended",
    "prettier"
  ],
  "plugins": ["react", "flowtype", "promise"],
  "rules": {
    "no-unused-expressions": "error",
  }
}
```
